core/shared/src/main/scala/matryoshka/Recursive.scala

```diff
@@ -184,66 +233,278 @@ import simulacrum._
+  def all(t: T)(p: T ⇒ Boolean)(implicit BF: Functor[Base], B: Foldable[Base]): Boolean =
     Tag.unwrap(foldMap(t)(p(_).conjunction))
 
-  def any[F[_]: Functor: Foldable](t: T[F])(p: T[F] ⇒ Boolean): Boolean =
+  def any(t: T)(p: T ⇒ Boolean)(implicit BF: Functor[Base], B: Foldable[Base]): Boolean =
     Tag.unwrap(foldMap(t)(p(_).disjunction))
 
-  def collect[F[_]: Functor: Foldable, B](t: T[F])(pf: PartialFunction[T[F], B]):
-      List[B] =
+  def collect[B]
+    (t: T)
+    (pf: PartialFunction[T, B])
+    (implicit BF: Functor[Base], B: Foldable[Base])
+      : List[B] =
     foldMap(t)(pf.lift(_).toList)
 
-  def contains[F[_]: Functor: Foldable](t: T[F], c: T[F])(implicit T: Equal[T[F]]):
-      Boolean =
+  def contains
+    (t: T, c: T)
+    (implicit T: Equal[T], BF: Functor[Base], B: Foldable[Base])
+      : Boolean =
     any(t)(_ ≟ c)
 
-  def foldMap[F[_]: Functor: Foldable, Z: Monoid](t: T[F])(f: T[F] => Z): Z =
-    foldMapM[F, Free.Trampoline, Z](t)(f(_).pure[Free.Trampoline]).run
-
-  def foldMapM[F[_]: Functor: Foldable, M[_]: Monad, Z: Monoid](t: T[F])(f: T[F] => M[Z]):
-      M[Z] = {
-    def loop(z0: Z, term: T[F]): M[Z] = {
+  def foldMap[Z: Monoid]
+    (t: T)
+    (f: T => Z)
+    (implicit BF: Functor[Base], B: Foldable[Base])
+      : Z =
+    foldMapM[Free.Trampoline, Z](t)(f(_).pure[Free.Trampoline]).run
+
+  def foldMapM[M[_]: Monad, Z: Monoid]
+    (t: T)
+    (f: T => M[Z])
+    (implicit BF: Functor[Base], B: Foldable[Base])
+      : M[Z] = {
+    def loop(z0: Z, term: T): M[Z] = {
       for {
         z1 <- f(term)
-        z2 <- project[F](term).foldLeftM(z0 ⊹ z1)(loop(_, _))
+        z2 <- project(term).foldLeftM(z0 ⊹ z1)(loop(_, _))
       } yield z2
     }
 
     loop(Monoid[Z].zero, t)
   }
 
-  def convertTo[F[_]: Functor, R[_[_]]: Corecursive](t: T[F]): R[F] =
-    cata(t)(Corecursive[R].embed[F])
+  def convertTo[R]
+    (t: T)
+    (implicit R: Corecursive.Aux[R, Base], BF: Functor[Base])
+      : R =
+    cata[R](t)(R.embed(_))
 }
 
 object Recursive {
-  def equalT[T[_[_]]](implicit T: Recursive[T]): EqualT[T] =
-    new EqualT[T] {
-      def equal[F[_]: Functor](tf1: T[F], tf2: T[F])(implicit del: Delay[Equal, F]) =
-        del(equalT[F](del)).equal(T.project(tf1), T.project(tf2))
-    }
+  type Aux[T, F[_]] = Recursive[T] { type Base[A] = F[A] }
+
+  def equal[T, F[_]: Functor]
+    (implicit T: Recursive.Aux[T, F], F: Delay[Equal, F])
+      : Equal[T] =
+    Equal.equal((a, b) => F(equal[T, F]).equal(T.project(a), T.project(b)))
+
+  def show[T, F[_]: Functor](implicit T: Recursive.Aux[T, F], F: Delay[Show, F])
+      : Show[T] =
+    Show.show(T.cata(_)(F(Cord.CordShow).show))
+
+  // NB: The rest of this is what would be generated by simulacrum, except this
+  //     type class is too complicated to take advantage of that.
+
+  def apply[T, F[_]](implicit instance: Aux[T, F]): Aux[T, F] = instance
+
+  trait Ops[T, F[_]] {
+    def typeClassInstance: Aux[T, F]
+    def self: T
+
```

Comment:
So, this code is literally copy/pasted from the Simulacrum expansion, with minimum changes to make it work. We can definitely make it cleaner now that it's here (especially these ops that are liable to stick around for a while). But I'd like to make various changes to Simulacrum to make more of it disappear – the first being making it possible to “override” individual generated methods with hand-written ones, which @mpilquist has already said he would like to see happen.